data/types/bolus/pen/pen.go

```diff
@@ -0,0 +1,63 @@
+package pen
+
+import (
+	"github.com/tidepool-org/platform/data"
+	"github.com/tidepool-org/platform/data/types/bolus"
+	"github.com/tidepool-org/platform/structure"
+)
+
+const (
+	SubType = "pen" // TODO: Rename Type to "bolus/normal"; remove SubType
+
+	NormalMaximum = 100.0
+	NormalMinimum = 0.0
+)
+
+type Pen struct {
+	bolus.Bolus `bson:",inline"`
+
+	Normal *float64 `json:"normal,omitempty" bson:"normal,omitempty"`
+}
+
+func New() *Pen {
+	return &Pen{
+		Bolus: bolus.New(SubType),
+	}
+}
+
+func (n *Pen) Parse(parser structure.ObjectParser) {
+	if !parser.HasMeta() {
+		parser = parser.WithMeta(n.Meta())
+	}
+
+	n.Bolus.Parse(parser)
+
+	n.Normal = parser.Float64("normal")
+}
+
+func (n *Pen) Validate(validator structure.Validator) {
+	if !validator.HasMeta() {
+		validator = validator.WithMeta(n.Meta())
+	}
+
+	n.Bolus.Validate(validator)
+
+	if n.SubType != "" {
+		validator.String("subType", &n.SubType).EqualTo(SubType)
+	}
+
```

Comment:
```suggestion
```
Theses lines from "normal" bolus are not necessary.
If subType is not "pen" this code won't be executed anyway.

Reply:
That's true the way we call it through the bolus factory. 
But nothing else in the code is preventing it to happen.